test/models/page_test.rb

```diff
@@ -13,4 +13,32 @@ class PageTest < ActiveSupport::TestCase
     page = Page.create(url: 'some/path/to/a/page')
     assert_not(page.valid?, 'The page should be invalid because it has no domain')
   end
+
+  test 'page title should sync with title from version with most recent capture time' do
+    page = pages(:home_page)
+    assert_equal('Page One', page.title)
+
+    page.versions.create(title: 'Newest Version', capture_time: '2017-03-05T00:00:00Z')
+    page.reload
+    assert_equal('Newest Version', page.title, 'The page title should always sync against the title of the version with the most recent capture time')
+    refute(page.changed?, 'The page was left with unsaved changes')
+    page.versions.create(title: 'Older Version', capture_time: '2017-03-01T00:00:00Z')
+    page.reload
+    refute_equal('Older Version', page.title, 'The page title should not sync against the title of a newly created version with an older capture time')
+    refute(page.changed?, 'The page was left with unsaved changes')
+  end
+
+  test "page title should not sync with a version's title if it's nil or blank" do
+    page = pages(:home_page)
+    assert_equal('Page One', page.title)
+
+    page.versions.create(capture_time: '2017-03-05T00:00:00Z')
+    page.reload
+    assert_equal('Page One', page.title, 'The page title should not sync with the incoming version if it has a nil title')
+    refute(page.changed?, 'The page was left with unsaved changes')
```

Comment:
…I *think* this is pointless after you’ve called `reload` (it trashes all state on the model object). Not totally sure on that, though.

Reply:
I also looked into this, and you're right. The `reload` method _does_ in fact trash any existing state on the model object. Also interestingly, however, I discovered (after throwing a couple `binding.pry` calls between a few of these assertions) that we don't actually need to call `reload` to get the new state on these objects.

```ruby
binding.pry
page.versions.create(title: 'Newest Version', capture_time: '2017-03-05T00:00:00Z')
binding.pry
```

`page.changed?` returned `false` from inside both of those `binding.pry`s despite the `page` record having actually had it's `title` attribute updated. So, it would seem we don't need to worry about calling `reload` to confirm these changes.

Reply:
I think this is because you changed the callback to use `update`, which saves immediately, so you *should* get `page.changed? == false`. I definitely tested this with an assertion before you made the latest changes and it did in fact fail. I think it would be good to have that assertion to guard against any regression there.

Reply:
Makes sense to me. I can add it back in as a regression safeguard 👍